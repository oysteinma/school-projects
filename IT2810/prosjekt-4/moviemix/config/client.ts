import { ApolloClient, NormalizedCacheObject } from "@apollo/client";

import { cache } from "./cache";

/**
 * Setup Apollo Client
 */
export const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
  cache: cache,
  uri: "https://it2810-59s.idi.ntnu.no/graphql",
});
