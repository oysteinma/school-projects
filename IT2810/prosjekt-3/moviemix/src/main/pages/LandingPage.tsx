import Box from "@mui/material/Box";
import { useQuery } from "@apollo/client";
import { GET_MOVIES } from "../queries/query";
import Search from "../components/Search";
import MovieMixHeader from "../components/MovieMixHeader";
import ErrorMessage from "../components/ErrorMessage";
import CardView from "../components/CardView";
import { Filter } from "../components/Filter";
import { useState } from "react";
import { moviesVar } from "../App";

/**
 * Gets the movies from the database and fetches more when scrolling to the bottom
 * @returns the LandingPage page, which displays the main page of the app
 */

export default function LandingPage() {
  const [limit, setLimit] = useState<number>(12);
  const [genre, setGenre] = useState<string>("");
  const [sortRating, setSortRating] = useState<string>("DESC");
  const [sortPopularity, setSortPopularity] = useState<string>("DESC");
  const [filter, setFiltering] = useState<string>("bothEnabled");

  const { error, refetch, fetchMore } = useQuery(GET_MOVIES, {
    variables:
      filter === "bothEnabled"
        ? {
            genre: genre,
            offset: 0,
            limit: limit,
            sort: { popularity: sortPopularity, averageRating: sortRating },
          }
        : filter === "onlyRating"
        ? // Needs to have standard value for popularity since several movies have the same rating
          {
            genre: genre,
            offset: 0,
            limit: limit,
            sort: { popularity: "DESC", averageRating: sortRating },
          }
        : {
            // ONLY POPULARITY
            genre: genre,
            offset: 0,
            limit: limit,
            sort: { popularity: sortPopularity },
          },
  });
  refetch().then((r) => {
    moviesVar(r.data.movies);
  });

  if (error) return <ErrorMessage {...error} />;

  window.onscroll = function LoadMoreContent() {
    if (window.scrollY + window.innerHeight >= document.body.scrollHeight) {
      const currentLength = moviesVar().length;
      fetchMore({
        variables: { offset: currentLength, limit: 12 },
      }).then((r) => {
        setLimit(currentLength + r.data.movies.length);
      });
    }
  };

  return (
    <Box
      display={"flex"}
      id="container"
      flexDirection={"column"}
      alignItems={"center"}
      alignContent={"center"}
      width={"50vh"}
      mx={"auto"}
      marginTop={"5vh"}
      marginBottom={"5vh"}
      gap={"5vh"}
    >
      <MovieMixHeader title="MovieMix" />
      <Search />
      <Filter
        onGenreChange={setGenre}
        onRatingChange={setSortRating}
        onPopularityChange={setSortPopularity}
        onFilteringChange={setFiltering}
        onLimitChange={setLimit}
        sortRating={sortRating}
        sortPopularity={sortPopularity}
      />
      <CardView />
    </Box>
  );
}
