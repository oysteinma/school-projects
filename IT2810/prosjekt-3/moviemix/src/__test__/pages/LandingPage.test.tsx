import LandingPage from "../../main/pages/LandingPage";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { HashRouter as Router } from "react-router-dom";

/**
 * @tests The landing page component - namely that the component renders
 */

it("check if LandingPage renders succesfully", () => {
  const { container } = render(
    <MockedProvider>
      <Router>
        <LandingPage />
      </Router>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});
