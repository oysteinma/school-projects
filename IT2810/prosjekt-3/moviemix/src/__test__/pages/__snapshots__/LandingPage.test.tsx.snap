// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if LandingPage renders succesfully 1`] = `
<div>
  <div
    class="MuiBox-root css-15qmrul"
    id="container"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1nhhm1w-MuiTypography-root"
    >
      MovieMix
    </h1>
    <div
      class="MuiBox-root css-bnbs76"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
              style="width: 40vw; height: 50px; background: white; border-radius: 0;"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                id="searchbar"
                placeholder="Search"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kbyzs3-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-1f2qzn3"
    >
      <div
        class="css-e53awj-MuiStack-root"
      >
        <button
          aria-label="Sort by popularity"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1tzf8hi-MuiButtonBase-root-MuiButton-root"
          id="popularityFilter"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TrendingUpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
              />
            </svg>
          </span>
          Trending
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Sort by rating"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1tzf8hi-MuiButtonBase-root-MuiButton-root"
          id="ratingFilter"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowUpwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
          </span>
          Rating
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root css-0"
        >
          <div
            aria-label="Filter by genre"
            class="MuiFormControl-root css-1q3rvyc-MuiFormControl-root"
            id="filterGenre"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-4cbs5r-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Genres
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1bkmtuu-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Genre
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <div
          class="MuiFormGroup-root MuiFormGroup-row css-ewkjcj-MuiFormGroup-root"
          color="white"
          role="radiogroup"
        >
          <label
            aria-label="Filter by genre and rating"
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-r2k4q6-MuiFormControlLabel-root"
            id="filterBoth"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name=":r1:"
                type="radio"
                value="bothEnabled"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Enable Both
            </span>
          </label>
          <label
            aria-label="Filter by trending only"
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-r2k4q6-MuiFormControlLabel-root"
            id="filterOnlyTrending"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name=":r1:"
                type="radio"
                value="onlyPopularity"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Only Trending
            </span>
          </label>
          <label
            aria-label="Filter by rating only"
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-r2k4q6-MuiFormControlLabel-root"
            id="filterOnlyRating"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name=":r1:"
                type="radio"
                value="onlyRating"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Only Rating
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
      style="width: 80vw;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10 css-vlkmuj-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
